feat(lean): created script to add a new step between two existing steps (v7)

RandellDawson opened the pull request:
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] All the files I changed are in the same world language, for example: only English changes, or only Chinese changes, etc.

This PR adds a new tool `create-step-between.js` that adds a new step between existing steps without having to manually create a new file or update the `meta.json` file.   Also made the code of all the functions a bit more DRY.  Lastly, updated `README.md` to explain how to use the new script works as well as `create-next-step.js`.

RandellDawson committed:
feat: create script to add step between two steps

RandellDawson committed:
fix: get curriculum path relative to tools folder

ojeytonwilliams commented:
As I mentioned in discussion with Randy, since this is entirely developer facing, I'm fine with this going in without tests.  It'll be obvious when you use it if there are problems.

I've blocked it just because next-curriculum is going away and I don't want to lose this. Once next-curriculum is in, we can change the base branch to master and I'll fix any conflicts if they arise.

ojeytonwilliams commented on curriculum/tools/README.md:
````diff
@@ -26,6 +26,17 @@ A one-off script that will automatically add a specified numbers of steps at a s
+### How to run script:
+1. Change to the directory of the project.
+2. Run the following npm command:
+  ```bash
+  npm run create-step-between start=X end=Y # where X is the starting step number and Y is the following step number.
+
````
````suggestion
  ```bash
  npm run create-step-between start=X end=Y # where X is the starting step number and Y is the following step number.
  ```
````

RandellDawson committed:
fix: added triple backticks to close code section

Co-authored-by: Oliver Eyton-Williams <ojeytonwilliams@gmail.com>

scissorsneedfoodtoo commented:
@RandellDawson, sorry for the late review.

Pulled this down locally and added some steps between projects. Everything LGTM -- this works great with the existing scripts.

RandellDawson commented:
@raisedadead @scissorsneedfoodtoo I have now targeted the `master` branch, so this can be merged now?

raisedadead commented:
We would want to move this up to the tools directory.

RandellDawson commented:
> We would want to move this up to the tools directory.

Eventually, but for now I just wanted to put it where the other scripts are located, so that I can use it.  That would be better in a separate issue?

raisedadead commented:
I can see that the scripts in that path target the v7, so we might as well take it up now? Also, that README file is the incorrect way writing out freeCodeCamp.

![image](https://user-images.githubusercontent.com/1884376/93396524-72ea6180-f895-11ea-9c0a-c47517ee5753.png)

RandellDawson commented:
> I can see that the scripts in that path target the v7, so we might as well take it up now?

Just tell me exactly where to move them and I can do it all on this PR.

> Also, that README file is the incorrect way writing out freeCodeCamp.

That is because this was originally a repo in my own GitHub account that I was using to help me create the steps.  I can make the necessary changes to the README.md

raisedadead commented:
`https://github.com/freeCodeCamp/freeCodeCamp/tree/master/tools/challenge-helper-scripts`

RandellDawson committed:
chore: move to tools/challenge-helper-scripts

RandellDawson commented:
@raisedadead After I figured how how to rewritie the packge.json files, I was able to move the scripts to the new location.  I tested the scripts and they all still work.  I am a little confused why the GitHub changes for the last commit showed deleting the `create-emtpy-steps.js` and creating a new one in the new folder instead of just renaming like the other two files, but that is just Git magic I supposed.

**Note:** I removed the two package that I added from the `curriculum` folder's `package.json` and created a new `package.json` in the new folder which included them.  I ran `npm install` in the new folder to generate the `package-lock.json` and finally updated the `lerna.json` file to include the new folder.

I believe I did everything correctly.  Just let me know if there is anything else I need to change.

scissorsneedfoodtoo commented:
Pulled this down locally and tested all the scripts. Everything LGTM :+1: :+1:
